### ISSUE TITLE
shows multiple lines of data for each frame

### ISSUE THREAD

**Issue by phatpaul** [ma-lwa-re/logic-lin-data#1]
I'm seeing multiple lines of data for each LIN frame.  Looks like it is appending a new byte each time.

![image](https://user-images.githubusercontent.com/3495287/220428293-d5404ec0-ebef-4a66-a420-057dceee6cd6.png)

```
name	type	start_time	duration	protected_id	data	index	checksum
LIN Data	header_pid	1.78085264	0.00049476	0x12			
LIN Data	header_pid	1.79087168	0.00049476	0x23			
LIN Data	header_pid	1.80102168	0.00049476	0x13			
LIN Data	header_pid	1.84890172	0.00049476	0x24			
LIN Data	header_pid	1.85087776	0.00049476	0x10			
LIN Data	header_pid	1.86109476	0.00049476	0x14			
LIN Data	header_pid	1.8710378	0.00049476	0x3C			
LIN Data	data	1.87949584	0.00049476	0x3C	0x01	0x0000000000000000	
LIN Data	data	1.87949584	0.00101568	0x3C	0x01, 0x06	0x0000000000000001	
LIN Data	data	1.87949584	0.00153568	0x3C	0x01, 0x06, 0xB2	0x0000000000000002	
LIN Data	data	1.87949584	0.00205672	0x3C	0x01, 0x06, 0xB2, 0x01	0x0000000000000003	
LIN Data	data	1.87949584	0.00257668	0x3C	0x01, 0x06, 0xB2, 0x01, 0xFF	0x0000000000000004	
LIN Data	data	1.87949584	0.00309776	0x3C	0x01, 0x06, 0xB2, 0x01, 0xFF, 0x7F	0x0000000000000005	
LIN Data	data	1.87949584	0.00361868	0x3C	0x01, 0x06, 0xB2, 0x01, 0xFF, 0x7F, 0xFF	0x0000000000000006	
LIN Data	data	1.87949584	0.00413876	0x3C	0x01, 0x06, 0xB2, 0x01, 0xFF, 0x7F, 0xFF, 0xFF	0x0000000000000007	
LIN Data	checksum	1.88366084	0.00049476				0x89
LIN Data	header_pid	1.89107976	0.00049476	0x3D			
LIN Data	data	1.89169488	0.00049476	0x3D	0x01	0x0000000000000000	
LIN Data	data	1.89169488	0.00101564	0x3D	0x01, 0x05	0x0000000000000001	
LIN Data	data	1.89169488	0.00153664	0x3D	0x01, 0x05, 0xF2	0x0000000000000002	
LIN Data	data	1.89169488	0.00205776	0x3D	0x01, 0x05, 0xF2, 0x98	0x0000000000000003	
LIN Data	data	1.89169488	0.00257868	0x3D	0x01, 0x05, 0xF2, 0x98, 0x49	0x0000000000000004	
LIN Data	data	1.89169488	0.0030996	0x3D	0x01, 0x05, 0xF2, 0x98, 0x49, 0x1E	0x0000000000000005	
LIN Data	data	1.89169488	0.00362072	0x3D	0x01, 0x05, 0xF2, 0x98, 0x49, 0x1E, 0x00	0x0000000000000006	
LIN Data	data	1.89169488	0.0041416	0x3D	0x01, 0x05, 0xF2, 0x98, 0x49, 0x1E, 0x00, 0xFF	0x0000000000000007	
LIN Data	checksum	1.8958628	0.00049476				0x89
```

Software is Logic 2.4.6
OS is xubuntu 20.04

**Comment by ma-lwa-re** [comment]
Hello!

First of all thank you for your feedback. Actually this came with the last update ([v1.0.2](https://github.com/ma-lwa-re/logic-lin-data/releases/tag/v1.0.2)) as I wanted to support all the possible LIN Data sizes.

In fact, the LIN specification allows variable length for data transfer. Most of the case, like in your example, 8 bytes are transferred followed by a checksum, but I had cases where only 4 bytes were used.

By appending a new value each time, I was sure to cover all possible length. However, it degrades the output quality.

I've come up with a fix, with the ability to configure the desired length (8 will be the default value) so this should produce a better experience. The new version will be released in the coming days after some proper testing.

Greetings!

**Comment by ma-lwa-re** [comment]
Fix available in the newest release [v1.0.3](https://github.com/ma-lwa-re/logic-lin-data/releases/tag/v1.0.3).

Update the plugin via the extension tab in the Logic app.
